server/tilesource/tiff_reader.py

```diff
@@ -170,6 +173,10 @@ def _validate(self):
 
         :raises: ValidationTiffException
         """
+        if not self._mustBeTiled:
+            if self._tiffInfo.get('istiled'):
+                raise ValidationTiffException('Expected a non-tiled TIFF file')
+            return
```

Comment:
This doesn't handle the `None` or `True` cases described in the docstring.

Reply:
If `_mustBeTiled` is truthy, it won't take this branch, and will check later (line 219).  I'll fix the `None` case.